internal/controllers/reconciliation/controller.go

```diff
@@ -402,7 +402,7 @@ func (c *Controller) requeue(logger logr.Logger, resource *resource.Snapshot, re
 		return ctrl.Result{RequeueAfter: wait.Jitter(c.readinessPollInterval, 0.1)}, nil
 	}
 
-	if resource == nil || (resource.Deleted() && !resource.Disable) || resource.ReconcileInterval == nil {
+	if resource == nil || resource.ReconcileInterval == nil {
```

Comment:
Why is the second term no longer required?

Reply:
It turns out this change didn't completely make sense. It fixed the foreground deletion test by requeueing  in cases where arguably we shouldn't.

I pushed a commit to keep the current logic, but add a special case: use the readiness polling interval when requeueing during foreground deletion.